core/templates/pages/contributor-dashboard-page/translation-language-selector/review-translation-language-selector.component.spec.ts

```diff
@@ -97,209 +107,241 @@ describe('Review Translation language selector', () => {
+      expect(component.languageIdToDescription.fr).toBe('français (French)');
+    });
 
-    clickDropdown();
-    expect(component.dropdownShown).toBe(true);
-    expect(getDropdownOptionsContainer()).toBeTruthy();
+    it('should correctly initialize dropdown activeLanguageCode', () => {
+      const dropdown = (
+        fixture.nativeElement.querySelector(
+          '.oppia-review-translation-language-selector-inner-container'));
 
-    clickDropdown();
-    expect(component.dropdownShown).toBe(false);
-    expect(getDropdownOptionsContainer()).toBeFalsy();
+      expect(dropdown.firstChild.textContent.trim()).toBe('English');
+    });
 
-    clickDropdown();
-    expect(component.dropdownShown).toBe(true);
-    expect(getDropdownOptionsContainer()).toBeTruthy();
+    it('should only show language options that the user can review' +
+      ' translations', () => {
+      expect(component.options.map(opt => opt.id)).toEqual(
+        translationReviewerLanguageCodes);
+      expect(component.filteredOptions).toEqual(component.options);
+    });
 
-    let fakeClickAwayEvent = new MouseEvent('click');
-    Object.defineProperty(
-      fakeClickAwayEvent,
-      'target',
-      {value: document.createElement('div')});
-    component.onDocumentClick(fakeClickAwayEvent);
-    fixture.detectChanges();
-    expect(component.dropdownShown).toBe(false);
-    expect(getDropdownOptionsContainer()).toBeFalsy();
-  }));
+    it('should correctly show and hide the dropdown', fakeAsync(() => {
+      expect(component.dropdownShown).toBe(false);
+      expect(getDropdownOptionsContainer()).toBeFalsy();
 
-  it('should correctly select and indicate selection of an option', () => {
-    spyOn(component.setActiveLanguageCode, 'emit');
+      clickDropdown();
+      expect(component.dropdownShown).toBe(true);
+      expect(getDropdownOptionsContainer()).toBeTruthy();
 
-    component.selectOption('fr');
-    fixture.detectChanges();
+      clickDropdown();
+      expect(component.dropdownShown).toBe(false);
+      expect(getDropdownOptionsContainer()).toBeFalsy();
 
-    expect(component.setActiveLanguageCode.emit).toHaveBeenCalledWith('fr');
-  });
+      clickDropdown();
+      expect(component.dropdownShown).toBe(true);
+      expect(getDropdownOptionsContainer()).toBeTruthy();
 
-  it('should display the selected language when the language is already' +
-    ' selected', () => {
-    component.activeLanguageCode = 'en';
+      let fakeClickAwayEvent = new MouseEvent('click');
+      Object.defineProperty(
+        fakeClickAwayEvent,
+        'target',
+        {value: document.createElement('div')});
+      component.onDocumentClick(fakeClickAwayEvent);
+      fixture.detectChanges();
+      expect(component.dropdownShown).toBe(false);
+      expect(getDropdownOptionsContainer()).toBeFalsy();
+    }));
 
-    component.ngOnInit();
+    it('should correctly select and indicate selection of an option', () => {
```

Comment:
Again, "correctly" is pretty vague, we can be more specific/clear, e.g.
```suggestion
    it('should set active language code to selected option', () => {
```